### PR TITLE
[Fixes #209] Get list of users

### LINKED ISSUE #209: Get list of users
## Feature Request

Is your feature request related to an existing [Issue](https://www.github.com/Auto-DL/Generator/issues) or a new implementation?

NO

### Feature Description

Users created on a local instance is required

### Possible Solution

os.listdir(user.rootpath)

### PR DESCRIPTION
**Pull request by TraXIcoN**
## Pull Request

fixes #209 

### Description
Create a method for getting all the users created on a local instance

### Checklist:
- [x] Followed the guidelines mentioned in our [Contributing document](../CONTRIBUTING.md)
- [x] Ensured that there aren't other open [Pull Requests](https://www.github.com/Auto-DL/Generator/pulls) for the same update/change
- [x] This PR is linked to an [Issue](https://www.github.com/Auto-DL/Generator/issues) and the issue number has been mentioned in the title and body of the PR
- [x] The code is clean and docstrings have been added or updated as required
- [x] Code is linted/formatted locally prior to submission (using `black`)

### CODE REVIEW

File: BackEndApp/v1/views.py [hunk 1]
```diff
@@ -494,3 +494,32 @@ def download_code(request):
...
+
+@api_view(["POST"])
+@is_authenticated
+def get_users(request):
+    try:
+        username = request.data.get("username")
+        user = User(username=username, password=None)
+        user = user.find()
+
+        users = os.listdir(os.path.expanduser("~") + "\.autodl")
```
Comment (ADI10HERO) [inline BackEndApp/v1/views.py:507]:
```suggestion
        users = os.listdir(user.rootpath)
```
Reason being if in the future we change the root path of the project from ~/.autodl to someplace else (Object/block storage on the internet) then we would need to make changes at multiple places :)

**Review by ADI10HERO — CHANGES_REQUESTED** [review]
Rest everything looks good.
I hope my suggestion makes sense to you as well.

File: BackEndApp/v1/views.py (hunk 1 above)
Comment (TraXIcoN) [reply BackEndApp/v1/views.py]: I Made the requested changes, check it out.

File: BackEndApp/v1/views.py [hunk 2]
```diff
@@ -501,15 +501,15 @@ def download_code(request):
 def get_users(request):
     try:
         username = request.data.get("username")
-        user = User(username=username, password=None)
+        user = User(username="asd", password=None)
```
Comment (ADI10HERO) [inline BackEndApp/v1/views.py:504]: Did you push right after testing?

File: BackEndApp/v1/views.py [hunk 3]
```diff
@@ -501,15 +501,15 @@ def download_code(request):
...
         user = user.find()
-
-        users = os.listdir(os.path.expanduser("~") + "\.autodl")
+        users = Store(user)
```
Comment (ADI10HERO) [inline BackEndApp/v1/views.py:506]:
```suggestion
        store = Store(user)
```

Naming of variables is one of the hardest things in software engineering, users = Store(user) doesn't sound right now, does it?

File: BackEndApp/v1/views.py [hunk 4]
```diff
@@ -501,15 +501,15 @@ def download_code(request):
...
+        users = os.listdir(users.rootpath)
```
Comment (ADI10HERO) [inline BackEndApp/v1/views.py:507]:
```suggestion
        users = os.listdir(store.rootpath)
```

File: BackEndApp/v1/views.py [hunk 5]
```diff
@@ -501,15 +501,15 @@ def download_code(request):
...
 
         status, success, message, data = 200, True, "Users fetched", users
```
Comment (ADI10HERO) [inline BackEndApp/v1/views.py:509]: `data` variable is redundant here, you can directly use `users` while sending

File: BackEndApp/v1/views.py [hunk 6]
```diff
@@ -501,15 +501,15 @@ def download_code(request):
...
 
     except Exception as e:
-        status, success, message, user = (
+        status, success, message, data = (
```
Comment (ADI10HERO) [inline BackEndApp/v1/views.py:512]: Here you're using `data = {}`
We need `data = []`
Even though it is empty, react code would expect a list not an "object"

**Review by ADI10HERO — CHANGES_REQUESTED** [review]

**Comment by TraXIcoN** [comment]
Whoa, I am just realising how clumsy I am, so sorry I will sort this mess right away!

**Comment by ADI10HERO** [comment]
> Whoa, I am just realising how clumsy I am, so sorry I will sort this mess right away!

I wouldn't say you're clumsy, these were genuine slips, it's okay, what's important is you're realizing and learning, and learnings are what Internships are for! 🎉

**Review by ADI10HERO — APPROVED** [review]

**Pull request merged**